### ISSUE TITLE
messages ResourceBundle is not loaded in modular apps

### ISSUE THREAD

**Issue by Burtan** [edvin/tornadofx#1286]
Hey,
when using the messages bundle in an modular app, it won't load with the following error:

```
java.lang.UnsupportedOperationException: ResourceBundle.Control not supported in named modules
```

**Comment by SchweinchenFuntik** [comment]
what version of TornadoFX? For Java 9+ you need to use 2.0.0-SNAPSHOT

**Comment by Burtan** [comment]
Sorry for not mentioning, I'm already using 2.0.0-SNAPSHOT.

**Comment by SchweinchenFuntik** [comment]
does your application use modules? I remember I had problems with modules and kotlin in gradle

**Comment by Burtan** [comment]
The application itself is modular, yes (with module-info.class). I'm using gradle 6.6.1 with this extension: https://github.com/jjohannes/extra-java-module-info. It builds fine. I'm pretty sure that TornadoFX is not using a valid method, thus the exception described in the first post.

**Comment by SchweinchenFuntik** [comment]
give a full stacktrace.

**Comment by Burtan** [comment]
In the class Component.kt of TornadoFX the error is thrown here:

```
    private val _messages: SimpleObjectProperty<ResourceBundle> = object : SimpleObjectProperty<ResourceBundle>() {
        override fun get(): ResourceBundle? {
            if (super.get() == null) {
                try {
                    val bundle = ResourceBundle.getBundle(FX.messagesNameProvider(this@Component.javaClass), FX.locale, this@Component.javaClass.classLoader, FXResourceBundleControl)
                    (bundle as? FXPropertyResourceBundle)?.inheritFromGlobal()
                    set(bundle)
                } catch (ex: Exception) {
                    FX.log.fine("No Messages found for ${javaClass.name} in locale ${FX.locale}, using global bundle")
                    set(FX.messages)
                }
            }
            return super.get()
        }
    }
```
The stacktrace is null.
```
detailMessage = "ResourceBundle.Control not supported in named modules"
walkback = {long[39]@2389} 
cause = {UnsupportedOperationException@2364} "java.lang.UnsupportedOperationException: ResourceBundle.Control not supported in named modules"
stackTrace = null
suppressedExceptions = {Collections$EmptyList@2390}  size = 0
disableWritableStackTrace = false
```
Do you want a minimal example on Github? I can upload one. I'm using AdoptOpenJDK 11.0.8. Tried on linux and windows.

**Comment by Burtan** [comment]
I hope [this ](https://github.com/Burtan/TornadoFxResourceBundleBug)helps

**Comment by Burtan** [comment]
[ResourceBundle.Control](https://github.com/edvin/tornadofx/blob/master/src/main/java/tornadofx/Messages.kt#L23) in Messages.kt has to be replaced when [using modules](https://docs.oracle.com/javase/9/docs/api/java/util/ResourceBundle.html).

**Comment by Burtan** [comment]
One more thing about resources that will be important. Using the old javaClass.getResource* can only load resources from the calling module. They way TornadoFx, the calling module is almost always not the module that contains the resources. E.g.

```
root = toolbar {
            imageview("/images/filenew.png") //the calling module is derived from "this@Toolbar" which is javafx.*
        }
```

The only way now resources like the image url work is to open the package that contains the resource in module-info.java. In this case this would be `opens images;` as described [here](https://docs.oracle.com/javase/9/docs/api/java/lang/Module.html#getResourceAsStream-java.lang.String-). I think this pitfall should be documented.

**Comment by Burtan** [comment]
Any update on this? Pull request wanted?

**Comment by Burtan** [comment]
What is the right branch to make a pull request from?

**Comment by SchweinchenFuntik** [comment]
https://github.com/edvin/tornadofx2

**Comment by Burtan** [comment]
Is tornadof2 available via maven anywhere? I guess the tornadofx:2.0.0-SNAPSHOT is still based on the tornadofx1 repo?

**Comment by SchweinchenFuntik** [comment]
just moved the `jdk 10` branch to the new repository

**Comment by Burtan** [comment]
Did I miss something? The repos seem to be still the same and snapshots wasn't updated?

**Comment by SchweinchenFuntik** [comment]
Alas, yes. I don't see any activity there. The project is inactive, abandoned. I don't know yet whether to do something or not.

**Comment by edvin** [comment]
Yeah, as I've mentioned I have no time to work on this, but others are of course welcome to contribute if time permits. I'll merge any reasonable PR's obviously. What we need is someone who has time and are willing to drive the project forward.

**Comment by SchweinchenFuntik** [comment]
@edvin here the question is rather about the second version, it is not clear what happened to it and in what condition it is
